dev/diffs/4.0.0.diff

```diff
@@ -71,62 +94,85 @@ index 466e4cf8131..798f118464c 100644
      val metadata = plan match {
        case fileScan: FileSourceScanLike => fileScan.metadata
 +      case cometScan: CometScanExec => cometScan.metadata
        case _ => Map[String, String]()
      }
-     new SparkPlanInfo(
+     val childrenInfo = children.flatMap {
+diff --git a/sql/core/src/test/resources/sql-tests/analyzer-results/listagg-collations.sql.out b/sql/core/src/test/resources/sql-tests/analyzer-results/listagg-collations.sql.out
+index 7aca17dcb25..8afeb3b4a2f 100644
+--- a/sql/core/src/test/resources/sql-tests/analyzer-results/listagg-collations.sql.out
++++ b/sql/core/src/test/resources/sql-tests/analyzer-results/listagg-collations.sql.out
+@@ -64,15 +64,6 @@ WithCTE
+       +- CTERelationRef xxxx, true, [c1#x], false, false
+ 
+ 
+--- !query
+-SELECT lower(listagg(DISTINCT c1 COLLATE utf8_lcase) WITHIN GROUP (ORDER BY c1 COLLATE utf8_lcase)) FROM (VALUES ('a'), ('B'), ('b'), ('A')) AS t(c1)
+--- !query analysis
+-Aggregate [lower(listagg(distinct collate(c1#x, utf8_lcase), null, collate(c1#x, utf8_lcase) ASC NULLS FIRST, 0, 0)) AS lower(listagg(DISTINCT collate(c1, utf8_lcase), NULL) WITHIN GROUP (ORDER BY collate(c1, utf8_lcase) ASC NULLS FIRST))#x]
+-+- SubqueryAlias t
+-   +- Project [col1#x AS c1#x]
+-      +- LocalRelation [col1#x]
+-
+-
+ -- !query
+ WITH t(c1) AS (SELECT replace(listagg(DISTINCT col1 COLLATE unicode_rtrim) COLLATE utf8_binary, ' ', '') FROM (VALUES ('xbc  '), ('xbc '), ('a'), ('xbc'))) SELECT len(c1), regexp_count(c1, 'a'), regexp_count(c1, 'xbc') FROM t
+ -- !query analysis
 diff --git a/sql/core/src/test/resources/sql-tests/inputs/collations.sql b/sql/core/src/test/resources/sql-tests/inputs/collations.sql
-index 619eb4470e9..8465382a007 100644
+index 17815ed5dde..baad440b1ce 100644
 --- a/sql/core/src/test/resources/sql-tests/inputs/collations.sql
 +++ b/sql/core/src/test/resources/sql-tests/inputs/collations.sql
-@@ -1,5 +1,8 @@
- -- test cases for collation support
- 
+@@ -1,3 +1,6 @@
 +-- TODO: https://github.com/apache/datafusion-comet/issues/551
 +--SET spark.comet.enabled = false
 +
+ -- test cases for collation support
+ 
  -- Create a test table with data
- create table t1(utf8_binary string collate utf8_binary, utf8_binary_lcase string collate utf8_binary_lcase) using parquet;
- insert into t1 values('aaa', 'aaa');
+diff --git a/sql/core/src/test/resources/sql-tests/inputs/decimalArithmeticOperations.sql b/sql/core/src/test/resources/sql-tests/inputs/decimalArithmeticOperations.sql
+index 13bbd9d81b7..fb6edc8ceb2 100644
+--- a/sql/core/src/test/resources/sql-tests/inputs/decimalArithmeticOperations.sql
++++ b/sql/core/src/test/resources/sql-tests/inputs/decimalArithmeticOperations.sql
+@@ -15,6 +15,10 @@
+ --   limitations under the License.
+ --
+ 
++-- TODO: Disabled due to one of the test failed for Spark4.0
++-- select /*+ COALESCE(1) */ id, a+b, a-b, a*b, a/b from decimals_test order by id
++--SET spark.comet.enabled = false
```

Comment:
Could you file a github issue and add a TODO link? E.g.
`+-- TODO: https://github.com/apache/datafusion-comet/issues/551`